ccm-connector/src/main/java/com/sequenceiq/cloudbreak/ccmimpl/ccmv2/CcmV2ManagementClient.java

```diff
@@ -0,0 +1,83 @@
+package com.sequenceiq.cloudbreak.ccmimpl.ccmv2;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.retry.backoff.FixedBackOffPolicy;
+import org.springframework.retry.policy.TimeoutRetryPolicy;
+import org.springframework.retry.support.RetryTemplate;
+import org.springframework.stereotype.Component;
+
+import com.cloudera.thunderhead.service.clusterconnectivitymanagementv2.ClusterConnectivityManagementV2Proto.UnregisterAgentResponse;
+import com.cloudera.thunderhead.service.clusterconnectivitymanagementv2.ClusterConnectivityManagementV2Proto.InvertingProxy;
+import com.cloudera.thunderhead.service.clusterconnectivitymanagementv2.ClusterConnectivityManagementV2Proto.InvertingProxyAgent;
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.ccmimpl.ccmv2.config.GrpcCcmV2Config;
+
+@Component
+public class CcmV2ManagementClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CcmV2ManagementClient.class);
+
+    @Inject
+    private GrpcCcmV2Client grpcCcmV2Client;
+
+    @Inject
+    private GrpcCcmV2Config grpcCcmV2Config;
+
+    public InvertingProxy awaitReadyInvertingProxyForAccount(String requestId, String accountId) {
+        return getRetryTemplate().execute(
+                retryContext -> {
+                    LOGGER.debug("Retrieving InvertingProxy Config for accountId '{}'", accountId);
+                    return Optional.ofNullable(grpcCcmV2Client.getOrCreateInvertingProxy(requestId, accountId, ThreadBasedUserCrnProvider.getUserCrn()))
+                            .filter(invertingProxy -> InvertingProxy.Status.READY.equals(invertingProxy.getStatus()))
+                            .orElseThrow(() -> new RuntimeException(String.format("InvertingProxy not found in ready state for accountId '%s'", accountId)));
```

Comment:
Ok introduced new exception. But there is not much handling the caller can do after ccm retry fails, the caller just rethrows the exception and stack creation is marked failure because of CCM issue. This behaviour is consistent with CCMV1.